lib/matplotlib/cm.py

```diff
@@ -371,6 +371,29 @@ def set_array(self, A):
             self._A = None
             return
 
+        if isinstance(self, mpl.collections.QuadMesh):
```

Comment:
Such a special-casing should not be done inside the base class. Instead, QuadMesh should override `set_array()`.

Reply:
Good catch - for further guidance, you can add the checks in `QuadMesh.set_array()` and then call `super().set_array()` to save duplicating any code....

Reply:
Yes, thanks for this :)